errors.go

```diff
@@ -34,6 +40,19 @@ func (e errAuthFailed) Is(target error) bool {
 	return ok
 }
 
+func (e errAuthFailed) ErrorCode() string {
+	errMsg := e.Inner.Error()
+
+	// Find the number in the format of ERR_NGROK_(\d+).
+	reg := regexp.MustCompile(`ERR_NGROK_(\d+)`)
+	matches := reg.FindStringSubmatch(errMsg)
+	if len(matches) == 2 {
+		return matches[1]
```

Comment:
It feels surprising that this returns, say, "123" instead of "ERR_NGROK_123".

The reason for that is consistency with the go api library: https://pkg.go.dev/github.com/ngrok/ngrok-api-go/v5#Error

There, we have `Error.ErrorCode`, which is `ERR_NGROK_123`, and then we have the helper method `ngrok.IsErrorCode(err, 400)`.

It feels like we should have consistency in how users think about errors, or at least have an explanation for why we're diverging if we do.

Reply:
> There, we have Error.ErrorCode, which is ERR_NGROK_123, and then we have the helper method ngrok.IsErrorCode(err, 400).

for some additional context as the author of ngrok-go-api: if i were to rev the ngrok api go client library, i'd change that function to be `ngrok.IsErrorCode(error, ...string)`. this further supports your suggestion. the reason being that 
we wanted to one day stop namespacing errors with numbers (all api errors are `ERR_NGROK_4XXX`) and instead move to namespacing with human-readable strings, e.g. `ERROR_NGROK_API_12` 

thanks for picking this up carl, this improvement is much appreciated!

Reply:
Oh, this might have been my bad. Carl is just porting my work in ngrok/ngrok-rust over to go, and I omitted the `ERR_NGROK_` prefix from the error code, since it seemed implied. I'm cool with adjusting it to have the prefix if we want to support namespacing in the future :+1: